google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplTest.java

```diff
@@ -620,6 +633,70 @@ public void testExecuteGetOptimizerVersion() {
     }
   }
 
+  @Test
+  public void testExecuteSetReturnCommitStats() {
+    try (ConnectionImpl subject =
+        createConnection(
+            ConnectionOptions.newBuilder()
+                .setCredentials(NoCredentials.getInstance())
+                .setUri(URI)
+                .build())) {
+      assertThat(subject.isReturnCommitStats(), is(equalTo(false)));
+
+      StatementResult res = subject.execute(Statement.of("set return_commit_stats=true"));
+      assertThat(res.getResultType(), is(equalTo(ResultType.NO_RESULT)));
+      assertThat(subject.isReturnCommitStats(), is(equalTo(true)));
+
+      res = subject.execute(Statement.of("set return_commit_stats=false"));
+      assertThat(res.getResultType(), is(equalTo(ResultType.NO_RESULT)));
+      assertThat(subject.isReturnCommitStats(), is(equalTo(false)));
+    }
+  }
+
+  @Test
+  public void testExecuteSetReturnCommitStatsInvalidValue() {
+    try (ConnectionImpl subject =
+        createConnection(
+            ConnectionOptions.newBuilder()
+                .setCredentials(NoCredentials.getInstance())
+                .setUri(URI)
+                .build())) {
+      assertThat(subject.isReturnCommitStats(), is(equalTo(false)));
+
+      try {
+        subject.execute(Statement.of("set return_commit_stats=yes"));
+        fail("Missing expected exception");
+      } catch (SpannerException e) {
+        assertThat(e.getErrorCode(), is(equalTo(ErrorCode.INVALID_ARGUMENT)));
```

Comment:
If you muse use Truth, then the way to write this is

assertThat(e.getErrorCode()).isEqualTo(ErrorCode.INVALID_ARGUMENT

but frankly you're better off not even using truth for basic equality, same as, null, non-null, and true and false comparisons. Truth is better reserved for more complex comparisons that JUnit doesn't naively support.

Reply:
This is actually not Truth, but [Hamcrest](https://github.com/googleapis/java-spanner/blob/b351d08d7567bc2a84213f1106b7e7f7fa56886e/google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionImplTest.java#L23). That was what this entire client library initially used, and most classes have gradually been migrated to Truth. This class has not.

I'll stick to JUnit for the simpler comparisons from now on.

(And changed this specific one to JUnit)